traveler-ios-ui/TravelerKitUI/Extensions/Facets+TravelerUI.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Facets+TravelerUI.swift
+//  TravelerKitUI
+//
+//  Created by Omar Padierna on 2019-09-06.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import Foundation
+import TravelerKit
+
+enum FacetType {
+    case prices
+    case category
+}
+
+extension Facets {
+    var availableFacets: [FacetType] {
```

Comment:
Because we have `Facet` as a model and now `FacetType` also as a model (in the UI Layer), I would call this `availableFacetTypes` so the return type not confusing.

The other thing here is that this is a computed property...imagine how many times the UITableViewDataSource will call this...An easy way around this is a ViewModel that is instantiated with the model and either calculates the available facets (or any other view attribute) at init or lazy loads it

```
struct MyNewViewModel { // And I hate calling things 'ViewModel', Im sure a better word exists
    var availableFacetTypes: [FacetType] = {
        // calculate
        return value
    }()

    init(facets: Facets) {
         self.facets = facets
    }
}
```